cmd/controller-manager/main.go

```diff
@@ -176,9 +177,11 @@ func main() {
 			deps.KubeInformerFactory,
 			deps.LabelFilterKubeInformerFactory,
 		}
+		waitCtx, cancel := context.WithTimeout(context.Background(), time.Second*30)
```

Comment:
@zhucan As you can see that in L176-L178, there are three informer factories including the ones list-watch the Kubernetes native resources, however, for some users, their Kubernetes cluster may have too many resources, in which case, we cannot sync all the resources in 30s, so we'd better add a startup parameter for tidb-controller-manager and make the default value as 30s.
You can refer to the code here https://github.com/pingcap/tidb-operator/blob/master/pkg/controller/dependences.go#L111.
Please also add a parameter in the helm chart https://github.com/pingcap/tidb-operator/blob/master/charts/tidb-operator/values.yaml#L42 in case that users want to customize this value.

Reply:
copy that, you want to make it configurable and default value as 30s?

Reply:
Yes, thanks!

Reply:
@zhucan It seems not changed as above comments

Reply:
the"resync-duration" can solve the issue as above comments. you mean I have no modify the yaml ?

Reply:
Resync duration is not prepared for this case. It is defined for interval of informer cache resync. I think a new option is needed.

Reply:
If use "resync-duration" in here, other codes using this flag will be affected